modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/security/IgniteSessionRepository.java

```diff
@@ -125,4 +137,34 @@ public IgniteSessionRepository setDefaultMaxInactiveInterval(int dfltMaxInactive
             throw convertToDatabaseNotAvailableException(e, messages.getMessage("err.db-not-available"));
         }
     }
+
+    /** {@inheritDoc} */
+    @Override public Map<String, ExpiringSession> findByIndexNameAndIndexValue(String idxName, String idxVal) {
+        if (!PRINCIPAL_NAME_INDEX_NAME.equals(idxName))
+            return Collections.emptyMap();
+
+        Collection<MapSession> sessions = new ArrayList<>();
+
+        try (QueryCursor<Cache.Entry<String, Object>> cursor = cache().query(new ScanQuery())) {
```

Comment:
Why doesn't use iterator?
```
stream(cache().spliterator(), false)
                .filter(item -> ...)
```

Reply:
I think this code should be in transaction block. This protect you from problems with client reconnect

Reply:
reworked to use index